src/NuGet.Core/NuGet.Commands/RestoreCommand/RequestFactory/DependencyGraphSpecRequestProvider.cs

```diff
@@ -151,9 +151,9 @@ private static ExternalProjectReference GetExternalProject(PackageSpec rootProje
             {
                 // Set properties from the restore metadata
                 ProjectStyle = project.PackageSpec.RestoreMetadata.ProjectStyle,
-                RestoreOutputPath = project.PackageSpec.RestoreMetadata.ProjectStyle == ProjectStyle.ProjectJson ? rootPath : project.PackageSpec.RestoreMetadata.OutputPath,
+                AssetsCachePath = project.PackageSpec.RestoreMetadata.ProjectStyle == ProjectStyle.ProjectJson ? rootPath : project.PackageSpec.RestoreMetadata.OutputPath,
                 DependencyGraphSpec = projectDgSpec,
-                BaseIntermediateOutputPath = projectPackageSpec.RestoreMetadata.OutputPath,
+                RestoreOutputPath = projectPackageSpec.RestoreMetadata.OutputPath,
```

Comment:
Should this be:
                RestoreOutputPath = project.PackageSpec.RestoreMetadata.ProjectStyle == ProjectStyle.ProjectJson ? rootPath : project.PackageSpec.RestoreMetadata.OutputPath,

Why did it change?

I feel like the assetscachepath and the restoreoutputpath conditions are swapped here, unless I am missing something. 

It seems to me that this is going to be null in the project.json case from commandline.

Reply:
I think you are right about the conditions being swapped.  Thanks for catching that!

> It seems to me that this is going to be null in the project.json case from commandline.

Looking at MSBuildRestoreUtility, it looks like the RestoreOutputPath will be set for project.json projects:

```C#
result.RestoreMetadata.OutputPath = specItem.GetProperty("OutputPath");
```